File: easy/python/max_profit.py
from ast import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max = 0
        price = prices[0]
        for x in prices:
            if max < x - price:
                max = x - price
            elif x < price:
                price = x
            print(max)
        return max

File: easy/ts/max_profit.ts
function maxProfit(prices: number[]): number {
  let maxProfit: number = 0;
  let buyDay: number = prices[0];
  for (let price of prices) {
    if (price < buyDay) {
      buyDay = price;
    }
    const profit = price - buyDay;
    if (maxProfit < profit) {
      maxProfit = profit;
    }
  }
  return maxProfit;
};